server/app/auth/ProfileUtils.java

```diff
@@ -35,6 +36,15 @@ public Optional<CiviFormProfile> currentUserProfile(Http.RequestHeader request)
     return currentUserProfile(webContext);
   }
 
+  /**
+   * Fetch the current profile from the session cookie, which the ProfileManager will fetch from the
+   * request's cookies, using the injected session store to decrypt it.
```

Comment:
suggestion (non-blocking): Worth adding a `@throws MissingOptionalException if we can't find the profile from the request` line to the Javadoc?

Reply:
The good ideas keep coming. Done!